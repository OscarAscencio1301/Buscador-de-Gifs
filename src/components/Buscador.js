import React, { useState } from 'react'

export const Buscador = ({setelementos}) => {
    const [valor, setvalor] = useState("")
    const cambioInput = (e) =>{
        setvalor(e.target.value)
        console.log("Se cambio el input")
    }
    const validarInput = (e) =>{
        e.preventDefault();
        setelementos(elementos => [valor,...elementos])
        setvalor("");
        console.log("Se activo el Submit")
    }
    return (
       
        <form onSubmit={validarInput}>
            <input 
                type="text"
                onChange={cambioInput}
                placeholder="Buscar..."
                value={valor}
            />
        </form>
      
    )
}
