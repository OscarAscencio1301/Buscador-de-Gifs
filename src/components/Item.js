import React from 'react'
import PropTypes from 'prop-types';

export const Item = ({imagenes}) => {
    return (
        <div className="item">
            <img src={imagenes.url} alt={imagenes.title}/>
            <div className="texto">
            <p>{imagenes.title}</p>
            </div>
        </div>
    )
}

Item.propTypes = {
    imagenes: PropTypes.object.isRequired
}