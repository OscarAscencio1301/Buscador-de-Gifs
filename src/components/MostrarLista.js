import React, { useState } from 'react'
import { consumirAPI } from '../helpers/consumirAPI';
import { useFetch } from '../hooks/useFetch';
import { Item } from './Item';

export const MostrarLista = ({elements}) => {
    // const [imagenes, setimagenes] = useState([])
    const {data, loading} = useFetch(elements);

    return (
        <div>
        {loading ? <h3 className="sub">Cargando...</h3> : null}
        <h3 className="titulo">{elements}</h3>
          <ol className="grid">
              {
                  data.map(imgs => 
                            <Item  
                                key={imgs.id}  
                                imagenes={imgs}/>)
              }
          </ol>
        </div>
    )
}
