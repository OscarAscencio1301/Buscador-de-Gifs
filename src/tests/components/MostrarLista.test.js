import { shallow } from "enzyme";
import { MostrarLista } from "../../components/MostrarLista";

describe('Pruebas al componente MostrarLista', () => {
    test('Realizando un snapshot', () => {
        const wrapper = shallow(<MostrarLista />)
        expect(wrapper).toMatchSnapshot();
    })

    test('Realizando un snapshot', () => {
        const elm = ["Batman" ,"goku"]
        const wrapper = shallow(<MostrarLista elements={elm}/>)
        expect(wrapper).toMatchSnapshot();
    })
    
});
