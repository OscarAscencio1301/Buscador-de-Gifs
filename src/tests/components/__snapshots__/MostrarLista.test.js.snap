// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas al componente MostrarLista Realizando un snapshot 1`] = `
<div>
  <h3
    className="sub"
  >
    Cargando...
  </h3>
  <h3
    className="titulo"
  />
  <ol
    className="grid"
  />
</div>
`;

exports[`Pruebas al componente MostrarLista Realizando un snapshot 2`] = `
<div>
  <h3
    className="sub"
  >
    Cargando...
  </h3>
  <h3
    className="titulo"
  >
    Batman
    goku
  </h3>
  <ol
    className="grid"
  />
</div>
`;
